docs2/site/docs/getting-started/errors.md

```diff
@@ -1,78 +1,203 @@
+exception's `Data` property.
+
+You can also handle these processing exceptions by setting a delegate within the
+`ExecutionOptions.UnhandledExceptionDelegate` property. Within the delegate you can log the error message
+and stack trace for debugging needs. You can also override the generic error message with a more specific
+message, wrap or replace the exception with your own `ExecutionError` class, and/or set the codes and data
+as necessary. Note that if `ThrowOnUnhandledExceptions` is `true`, the `UnhandledExceptionDelegate` will not be called.
+
+Here is a sample of a typical unhandled exception delegate which logs the error to a database.
+It also returns the log id along with the error message:
 
 ```csharp
-var executor = new DocumentExecutor();
-ExecutionResult result = await executor.ExecuteAsync(_ =>
+var executer = new DocumentExecuter();
+var result = executer.ExecuteAsync(options =>
 {
-  _.Query = "...";
-  _.UnhandledExceptionDelegate = context => context.ErrorMessage = "Error Message";
+
+    ...
+
+    options.UnhandledExecutionDelegate = context =>
+    {
+        try
+        {
+            using var db = new MyDatabaseContext();
+            var errorLog = new ErrorLog {
+                DateStamp = DateTime.UtcNow,
+                Message = context.Exception.Message,
+                Details = context.Exception.ToString()
+            };
+            db.ErrorLogs.Add(errorLog);
+            db.SaveChanges();
+            context.Exception.Data["errorLogId"] = errorLog.Id;
+        }
+        catch { }
+    };
 });
 ```
 
-Set `ExecutionOptions.ThrowOnUnhandledException` to `true` in cases where the thrown exception needs to propagate to an enclosing or global handler.
-
-Note that `UnhandledExceptionDelegate` will not be invoked in the case `ThrowOnUnhandledException` is `true`.
+You can also override the serialized exception by setting `context.Exception`, or simply replace
+the message by setting `context.ErrorMessage` such as in this example:
 
 ```csharp
-try
-{
-  var result = await executor.ExecuteAsync(_ =>
-  {
-    _.Query = "...";
-    _.ThrowOnUnhandledException = true;
-  });
-  
-  //Process result...
-}
-catch (Exception ex)
+options.UnhandledExecutionDelegate = ctx =>
 {
-  logger.Error(ex, "Error Message");
-}
+    if (ctx.Exception is SqlException)
+        ctx.ErrorMessage = "A database error has occurred.";
+};
 ```
 
-You can provide additional error handling or logging for fields by adding Field Middleware.
+## Error Serialization
+
+After the `DocumentExecuter` has returned a `ExecutionResult` containing the data and/or errors,
+typically you will pass this object to an implementation of `IDocumentWriter` to convert the
+object tree into json. The GraphQL spec allows for four properties to be returned with each
+error: `message`, `locations`, `path`, and `extensions`. The `IDocumentWriter` implementations
+provided for the `Newtonsoft.Json` and `System.Text.Json` libraries allow you to control the
```

Comment:
Maybe it is worth giving links to nuget packages?

Reply:
lol - I guess I did mash that sentence in here.

Reply:
Sure I can do that.

Reply:
Done